substrate/primitives/state-machine/src/basic.rs

```diff
@@ -119,8 +121,9 @@ impl BasicExternalities {
 	}
 }
 
-impl PartialEq for BasicExternalities {
-	fn eq(&self, other: &BasicExternalities) -> bool {
+impl BasicExternalities {
+	/// Same as `Eq` trait but on mutable references.
+	pub fn eq(&mut self, other: &mut BasicExternalities) -> bool {
 		self.overlay.changes().map(|(k, v)| (k, v.value())).collect::<BTreeMap<_, _>>() ==
 			other.overlay.changes().map(|(k, v)| (k, v.value())).collect::<BTreeMap<_, _>>() &&
```

Comment:
I think its possibly to directly do a comparison on the iterators instead of collecting them rist.

Reply:
yes this is mostly a test function, I will try to put it behind a cfg(test) or even remove it.